rest/functions_api.go

```diff
@@ -56,15 +60,26 @@ func (h *handler) handleFunctionCall() error {
 }
 
 // Subroutine for reading function name and arguments from a request
-func (h *handler) getFunctionArgs() (string, map[string]interface{}, error) {
+func (h *handler) getFunctionArgs(maxSize *int) (string, map[string]interface{}, error) {
 	name := h.PathVar(kFnNameParam)
 	args := map[string]interface{}{}
 	var err error
 	if h.rq.Method == "POST" {
 		// POST: Args come from the request body in JSON format:
+		if h.rq.ContentLength >= 0 {
+			if err := db.CheckRequestSize(h.rq.ContentLength, maxSize); err != nil {
+				return "", nil, err
+			}
+		}
 		err = h.readJSONInto(&args)
+		if err == nil && h.rq.ContentLength < 0 && maxSize != nil {
+			err = db.CheckRequestSize(int64(db.EstimateSizeOfJSON(args)), maxSize)
```

Comment:
It feels relatively cumbersome to estimate the size of JSON by iterating over elements in map[string]interface{}{}, when we just unmarshalled the raw bytes for that JSON a few lines earlier. 

json.Decoder has an InputOffset property that can be read post-Decode - could you use that to add a version of readJSONInto that returned bytes read?

Reply:
OK. It means using json.Decoder directly instead of going through those functions, but the other stuff they do is mostly not relevant, like checking for unknown fields. I did have to add the Content-Type check.

Q: The existing code uses base.JSONDecoder(), which returns an interface base.JSONDecoderI that does not support the InputOffset method. It looks like this is done to support pluggable JSON implementations?

Reply:
That's a good point - you may need to switch to using the native library directly, as the other implementation of JSONDecoderI doesn't have support for InputOffset (despite a pending PR https://github.com/json-iterator/go/pull/493).

Reply:
Yup; see my changes next time you review the PR.